packages/nextjs/components/scaffold-eth/Address/Address.tsx

```diff
@@ -0,0 +1,167 @@
+};
+
+const copyIconSizeMap = {
+  xs: "h-3.5 w-3.5",
+  sm: "h-4 w-4",
+  base: "h-[18px] w-[18px]",
+  lg: "h-5 w-5",
+  xl: "h-[22px] w-[22px]",
+  "2xl": "h-6 w-6",
+  "3xl": "h-[26px] w-[26px]",
+};
+
+const getNextSize = (currentSize: keyof typeof sizeMap, step = 1): keyof typeof sizeMap => {
+  const sizes = Object.keys(sizeMap) as Array<keyof typeof sizeMap>;
+  const currentIndex = sizes.indexOf(currentSize);
+  const nextIndex = Math.min(currentIndex + step, sizes.length - 1);
+  return sizes[nextIndex];
+};
+
+type AddressProps = {
+  address?: AddressType;
+  disableAddressLink?: boolean;
+  format?: "short" | "long";
+  size?: "xs" | "sm" | "base" | "lg" | "xl" | "2xl" | "3xl";
+  showBoth?: boolean;
+};
+
+export const Address = ({ address, disableAddressLink, format, size, showBoth = false }: AddressProps) => {
+  const [ens, setEns] = useState<string | null>();
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+  const checkSumAddress = address ? getAddress(address) : undefined;
+
+  const { targetNetwork } = useTargetNetwork();
+
+  const { data: fetchedEns } = useEnsName({
+    address: checkSumAddress,
+    chainId: 1,
+    query: {
+      enabled: isAddress(checkSumAddress ?? ""),
+    },
+  });
+  const { data: fetchedEnsAvatar } = useEnsAvatar({
+    name: fetchedEns ? normalize(fetchedEns) : undefined,
+    chainId: 1,
+    query: {
+      enabled: Boolean(fetchedEns),
+      gcTime: 30_000,
+    },
+  });
+
+  useEffect(() => {
+    setEns(fetchedEns);
+  }, [fetchedEns]);
+
+  useEffect(() => {
+    setEnsAvatar(fetchedEnsAvatar);
+  }, [fetchedEnsAvatar]);
+
+  if (!checkSumAddress) {
+    return (
+      <div className="animate-pulse flex space-x-4">
+        <div className="rounded-md bg-slate-300 h-6 w-6"></div>
+        <div className="flex items-center space-y-6">
+          <div className="h-2 w-28 bg-slate-300 rounded"></div>
+        </div>
+      </div>
+    );
+  }
+
+  if (!isAddress(checkSumAddress)) {
+    return <span className="text-error">Wrong address</span>;
+  }
+
+  const blockExplorerAddressLink = getBlockExplorerAddressLink(targetNetwork, checkSumAddress);
+  const shortAddress = checkSumAddress?.slice(0, 6) + "..." + checkSumAddress?.slice(-4);
+  const displayAddress = ens || (format === "long" ? checkSumAddress : shortAddress);
+
+  size = size ?? (showBoth && ens ? "xs" : "base");
+  const addressSize = size;
+  const blockieSize = showBoth && ens ? getNextSize(size, 4) : size;
+  const ensSize = showBoth && ens ? getNextSize(size) : size;
+
+  return (
+    <div className="flex items-center flex-shrink-0">
+      <div className="flex-shrink-0">
+        <BlockieAvatar
+          address={checkSumAddress}
+          ensImage={ensAvatar}
+          size={(blockieSizeMap[blockieSize] * 24) / blockieSizeMap["base"]}
+        />
+      </div>
+      {showBoth && ens ? (
+        <div className="flex flex-col">
+          <span className={`ml-1.5 text-${ensSize} font-bold`}>
```

Comment:
Ohh yeah I noticed that in `main` branch we had same and since we were using almost all text-sizes in different component it was working. But yeah that was because we were just lucky, fixed at https://github.com/scaffold-eth/scaffold-eth-2/pull/924/commits/b0844a1cf8c5ed766e034dd5be7c1cccfef6d656